confmap/resolver.go

```diff
@@ -12,8 +12,15 @@ import (
 
 	"go.uber.org/multierr"
 	"go.uber.org/zap"
+
+	"go.opentelemetry.io/collector/featuregate"
 )
 
+var UseUnifiedEnvVarExpansionRules = featuregate.GlobalRegistry().MustRegister("confmap.unifyEnvVarExpansion",
```

Comment:
Can we make it so that this is not a public symbol? (e.g. move to `confmap/internal`)

Reply:
@mx-psi we cannot, since `otelcol.NewCommand` depends on it.

Reply:
Can do